src/app/Library/CrudPanel/Traits/Validation.php

```diff
@@ -218,14 +296,75 @@ public function isRequired($inputKey)
         }
 
         if (Str::contains($inputKey, '.')) {
-            $entity_array = explode('.', $inputKey);
-            $name_string = '';
-            foreach ($entity_array as $arr_key => $array_entity) {
-                $name_string .= ($arr_key === 0) ? $array_entity : '['.$array_entity.']';
-            }
-            $inputKey = $name_string;
+            $inputKey = $this->getStringFromDotNotationToBrackets($inputKey);
         }
 
         return in_array($inputKey, $this->getOperationSetting('requiredFields'));
     }
+
+    /**
+     * Return the dot.notation.string as bracket[notation][string].
+     *
+     * @param  string  $string  field name or input name
+     * @return string
+     */
+    private function getStringFromDotNotationToBrackets(string $string)
+    {
+        $stringParts = explode('.', $string);
+        $result = '';
+
+        foreach ($stringParts as $key => $part) {
+            if ($part === '*') {
+                continue;
+            }
+            $result .= ($key === 0) ? $part : '['.$part.']';
+        }
+
+        return $result;
+    }
```

Comment:
I love the fact that you've split this into a separate function! Bravo 👏👏👏

Now that I see it as a separate function, I think it's obvious it doesn't belong inside the `Validation` trait - it's a general-purpose string manipulation function. So let's either 

(A) create a helper for it

Oooor... better yet... we could 

(B) add a Macro to Laravel's `Str` class, to be able do something like `Str::of('user.address.street')->dotsToSquareBrackets()`. Which I think is super-readable and in the right place. String manipulation function... on top of the string manipulation library. Right?